### PR TITLE
fix: resolving rate limit issue

### PR DESCRIPTION
**Pull request by harshilgajera-crest**
Fixed two issues:
- 429 rate limit
- Docker pull limit per hour

Reduced api calls from around ~100 to 1.
Redesigned the splunk_matrix_update.py file and added doc strings for better understanding.

### CODE REVIEW

**Comment by artemrys** [comment]
Incorporates https://github.com/splunk/addonfactory-test-matrix-action/pull/175.

File: splunk_matrix_update.py [hunk 1]
```diff
@@ -2,114 +2,151 @@
...
-        for i in range(len(filtered_images)):
-            filtered_images[i] = filtered_images[i].replace("'", "")
+        filtered_images = [image.replace("'", "") for image in filtered_images]
         filtered_images.sort(key=lambda s: list(map(int, s.split("."))))
         return filtered_images[-1]
+    return None
 
 
-def check_image_version(latest_image, stanza_image):
+def check_image_version(latest_image: str, stanza_image: str) -> bool:
```
[the remark this answers is not in the document]
Comment (harshilgajera-crest) [reply splunk_matrix_update.py:47]: Done.

File: splunk_matrix_update.py [hunk 2]
```diff
@@ -2,114 +2,151 @@
...
-
-                if check_image_version(latest_image_version, stanza_image_version):
-                    config.set(stanza, "VERSION", latest_image_version)
-                    latest_image_digest = get_image_digest(token, latest_image_version)
-                    build_number = get_build_number(
-                        token, filter_images, latest_image_digest
-                    )
-                    config.set(stanza, "BUILD", build_number)
-                    update_file = True
+                latest_image_version = get_latest_image(stanza, all_images_list)
```
[the remark this answers is not in the document]
Comment (harshilgajera-crest) [reply splunk_matrix_update.py:124]: Not sure what you mean by this, I believe current flow for this function in okay.
Both get_latest_image and check_image_version takes diff set of args.

**Pull request merged**